perl-lib/OESS/lib/OESS/DB/User.pm

```diff
@@ -387,9 +392,6 @@ sub get_workgroups {
     if (!defined $datas) {
         return (undef, $args->{db}->get_error);
     }
-    if (!defined $datas->[0]){
-        return (undef, "Returned 0 Workgroups");
-    }
```

Comment:
This is probably correct, but from what I've seen in other locations this check is made in the situation that no work groups exist but there isn't an error. This could be old vs new database library in which case I trust your knowledge of it. I just know I've see this check done presumably because the query executes without issue, but there is nothing to return.

Reply:
These types of checks should only raise an error if we expect at least one value. For example, when we query for a particular resource by id, it's expected that a singular resource would be returned.

In this case we're querying for a user's workgroups; Is it really an error if a user isn't in a workgroup? I'd argue no.